mitodl/open-discussions
# Minimum search query length

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #1507

#### What's this PR do?
- Sets a configurable minimum length for the search query parameter (default=2)
- Does not run the search if the query string is less than the minimum length.
- Shows an error message if the input is submitted and the query string is less than the minimum length.

#### How should this be manually tested?
- Under global and channel search, try submitting a search with 0, 1, and 2+ character search terms.
- You should see this with < 2-character searches:
![screen shot 2019-01-10 at 10 44 19 am](https://user-images.githubusercontent.com/187676/50979564-bf787980-14c4-11e9-8e6d-601484d775d9.png)
- You should see results (or a `There are no results to display.` message depending on query) for 2+ character searches.
